### PR TITLE
Remove remaining request builders used in IndicesAdminClient

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java [hunk 1]
```diff
@@ -350,9 +351,9 @@ protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener
         Settings readOnlySettings = readOnly ? Settings.builder()
             .put(IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE, Boolean.TRUE.toString()).build() :
             Settings.builder().putNull(IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE).build();
-        client.admin().indices().prepareUpdateSettings(indicesToUpdate.toArray(Strings.EMPTY_ARRAY))
-            .setSettings(readOnlySettings)
-            .execute(ActionListener.map(wrappedListener, r -> null));
+        client.admin().indices().updateSettings(new UpdateSettingsRequest(readOnlySettings, indicesToUpdate.toArray(Strings.EMPTY_ARRAY)))
+            .thenApply(response -> (Void) null)
```
Comment (matriv) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java:355]: Just curiosity, why is this needed?

File: server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesRequest.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public class GetIndexTemplatesRequest extends MasterNodeReadRequest<GetIndexTemp
     private String[] names;
 
     public GetIndexTemplatesRequest() {
+        this.names = new String[0];
```
Comment (matriv) [inline server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesRequest.java:37]: Maybe a comment to explain, what's happening in this case?
Do we get templates for all indices?

**Review by matriv — APPROVED** [review]
Thx! Left a couple of questions.

File: server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesRequest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesRequest.java:37]: Yes,  the RequestBuilder did this implicitly due to varargs usage. Using this constructor without calling .indices resulted in a NPE.

I guess an alternative would be to remove this ctor

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java:355]: Because the ActionListener is using `<Void>`. It doesn't type check without it
